test/routes/add-article.test.ts

```diff
@@ -23,10 +23,37 @@ describe('add article', (): void => {
 
     expect(response.status).toBe(204);
     expect(await articles.count()).toBe(1);
-    expect([...articles][0]['http://schema.org/name']).toEqual('Article');
+    expect([...articles][0]['http://schema.org/name']).toEqual([{ '@value': 'Article' }]);
   });
 
   it('should throw an error if id is already set', async (): Promise<void> => {
     await expect(makeRequest(createArticle('_:1'))).rejects.toThrow(new createHttpError.Forbidden('Article IDs must not be set (\'_:1\' was given)'));
   });
+
+  it('should throw an error if it is not a schema:Article', async (): Promise<void> => {
+    await expect(makeRequest({
+      '@type': 'http://schema.org/NewsArticle',
+      'http://schema.org/name': 'Article _:1',
+    })).rejects.toThrow(new createHttpError.BadRequest('Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)'));
```

Comment:
Was wondering why these were passing. Looks like `toThrow` only compares the message of the error, not the error itself...

Reply:
`toThrow(createHttpError.BadRequest)` seems to compare the type.

Reply:
Have you tried passing a callback rather than the error object? 
```
toThrow(() => new createHttpError.BadRequest('Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)'))```

Reply:
Doesn't take a function.

Wonder if https://www.npmjs.com/package/jest-extended#tothrowwithmessagetype-message should be used everywhere...

Reply:
Wonder if you can compare exceptions at all apart from separately looking at `message` and `name` (which is the type); two otherwise identical exceptions created in two different lines have a different `stack` property:
```
> [(new Error("42")).stack, (new Error("42")).stack]
[
  'Error: 42\n' +
    '    at repl:1:3\n' +
    '    at Script.runInThisContext (vm.js:116:20)\n' +
    '    at REPLServer.defaultEval (repl.js:404:29)\n' +
    '    at bound (domain.js:420:14)\n' +
    '    at REPLServer.runBound [as eval] (domain.js:433:12)\n' +
    '    at REPLServer.onLine (repl.js:715:10)\n' +
    '    at REPLServer.emit (events.js:215:7)\n' +
    '    at REPLServer.EventEmitter.emit (domain.js:476:20)\n' +
    '    at REPLServer.Interface._onLine (readline.js:316:10)\n' +
    '    at REPLServer.Interface._line (readline.js:693:8)',
  'Error: 42\n' +
    '    at repl:1:28\n' +
    '    at Script.runInThisContext (vm.js:116:20)\n' +
    '    at REPLServer.defaultEval (repl.js:404:29)\n' +
    '    at bound (domain.js:420:14)\n' +
    '    at REPLServer.runBound [as eval] (domain.js:433:12)\n' +
    '    at REPLServer.onLine (repl.js:715:10)\n' +
    '    at REPLServer.emit (events.js:215:7)\n' +
    '    at REPLServer.EventEmitter.emit (domain.js:476:20)\n' +
    '    at REPLServer.Interface._onLine (readline.js:316:10)\n' +
    '    at REPLServer.Interface._line (readline.js:693:8)'
]
```

Reply:
> Wonder if you can compare exceptions at all apart

Reply:
> Doesn't take a function.
> 
Ah, yeah I was thinking of chai. The API is so similar, the differences can be sneaky.

Reply:
Trying to setup `jest-extended` so I can use the `toThrowWithMessage` assertion. Having trouble with setup: https://github.com/libero/article-store/pull/101

Reply:
I usually use a `toEqual` assertion when using `.rejects`. Could this work? 
```
expect(async () => await makeRequest({
      '@type': 'http://schema.org/NewsArticle',
      'http://schema.org/name': 'Article _:1',
    })).toThrow(new createHttpError.BadRequest('Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)'));
```